### PR TITLE
chore: permissions and permission summary proto

### PR DESCRIPTION
**Pull request by NicholasBlaskey**
## Description

Permissions proto and permissions summary endpoint proto.

## Checklist
- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.
- [ ] If modifying `/webui/react/src/shared/` verify `make -C webui/react test-shared` passes.

### CODE REVIEW

File: proto/src/determined/rbac/v1/rbac.proto [hunk 1]
```diff
@@ -19,10 +19,74 @@ message Role {
...
+
+// List of permissions types.
+enum PermissionType {
+  // The permission type is unknown.
+  PERMISSION_TYPE_UNSPECIFIED = 0;
+
+  // Can create and update other users.
+  // Allows updating other users passwords making this
+  // permission give all other permissions effectively.
+  PERMISSION_TYPE_ADMINISTRATE_USER = 91001;
```
Comment (ioga) [inline proto/src/determined/rbac/v1/rbac.proto:37]: is 91001 intentional?
```suggestion
  PERMISSION_TYPE_ADMINISTRATE_USER = 1001;
```

File: proto/src/determined/rbac/v1/rbac.proto [hunk 2]
```diff
@@ -19,10 +19,74 @@ message Role {
...
+  PERMISSION_TYPE_VIEW_EXPERIMENT_METADATA = 2003;
+  // Ability to update experiment and experiment's lifecycle.
+  PERMISSION_TYPE_UPDATE_EXPERIMENT = 2004;
+  // Ability to update experiment's metadata.
+  PERMISSION_TYPE_UPDATE_EXPERIMENT_METADATA = 2005;
+  // Ability to delete experiment.
+  PERMISSION_TYPE_DELETE_EXPERIMENT = 2006;
+
+  // Ability to create, update, and add / remove users from groups.
+  PERMISSION_TYPE_UPDATE_GROUP = 93001;
```
Comment (ioga) [inline proto/src/determined/rbac/v1/rbac.proto:53]:
```suggestion
  PERMISSION_TYPE_UPDATE_GROUP = 3001;
```

File: proto/src/determined/rbac/v1/rbac.proto [hunk 3]
```diff
@@ -19,10 +19,74 @@ message Role {
...
+
+  // Ability to create workspaces.
+  PERMISSION_TYPE_CREATE_WORKSPACE = 94001;
```
Comment (ioga) [inline proto/src/determined/rbac/v1/rbac.proto:56]:
```suggestion
  PERMISSION_TYPE_CREATE_WORKSPACE = 4001;
```

File: proto/src/determined/rbac/v1/rbac.proto [hunk 4]
```diff
@@ -19,10 +19,74 @@ message Role {
...
+  PERMISSION_TYPE_CREATE_PROJECT = 5001;
+  // Ability to view projects.
+  PERMISSION_TYPE_VIEW_PROJECT = 5002;
+  // Ability to update projects.
+  PERMISSION_TYPE_UPDATE_PROJECT = 5003;
+  // Ability to delete projects.
+  PERMISSION_TYPE_DELETE_PROJECT = 5004;
+
+  // Ability to create and update role definitions.
+  PERMISSION_TYPE_UPDATE_ROLES = 96001;
```
Comment (ioga) [inline proto/src/determined/rbac/v1/rbac.proto:74]:
```suggestion
  PERMISSION_TYPE_UPDATE_ROLES = 6001;
```

File: proto/src/determined/rbac/v1/rbac.proto [hunk 5]
```diff
@@ -19,10 +19,74 @@ message Role {
...
+  PERMISSION_TYPE_CREATE_WORKSPACE = 94001;
+  // Ability to view workspace.
+  PERMISSION_TYPE_VIEW_WORKSPACE = 4002;
+  // Ability to update workspace.
+  PERMISSION_TYPE_UPDATE_WORKSPACE = 4003;
+  // Ability to delete workspace.
+  PERMISSION_TYPE_DELETE_WORKSPACE = 4004;
+
+  // Ability to create projects.
+  PERMISSION_TYPE_CREATE_PROJECT = 5001;
```
Comment (ioga) [inline proto/src/determined/rbac/v1/rbac.proto:65]: or should this be 95001? although I am not sure what's the case for that.

**Review by ioga — APPROVED** [review]

File: proto/src/determined/rbac/v1/rbac.proto (hunk 5 above)
Comment (NicholasBlaskey) [reply proto/src/determined/rbac/v1/rbac.proto:65]: 9xxxx refers to it being global scope only permission. Though I'll add a comment explaining it.

**Pull request merged**